mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -517,69 +516,75 @@ class WidgetAdapter(
         private val connection: Connection,
         colorMapper: ColorMapper
     ) : LabeledItemBaseViewHolder(inflater, parent, R.layout.widgetlist_slideritem, connection, colorMapper),
-        SeekBar.OnSeekBarChangeListener {
-        private val seekBar: SeekBar = itemView.findViewById(R.id.seekbar)
+        Slider.OnSliderTouchListener, Slider.LabelFormatter {
+        private val seekBar: Slider = itemView.findViewById(R.id.seekbar)
         private var boundWidget: Widget? = null
 
         init {
-            seekBar.setOnSeekBarChangeListener(this)
-            val now = Calendar.getInstance()
-            if (now.get(Calendar.DAY_OF_MONTH) == 31 && now.get(Calendar.MONTH) == Calendar.OCTOBER) {
-                seekBar.thumb = ContextCompat.getDrawable(itemView.context, R.drawable.ic_halloween_orange_24dp)
-            }
+            seekBar.addOnSliderTouchListener(this)
+            seekBar.setLabelFormatter(this)
         }
 
         override fun bind(widget: Widget) {
             super.bind(widget)
             boundWidget = widget
 
-            val stepCount = (widget.maxValue - widget.minValue) / widget.step
-            seekBar.max = Math.ceil(stepCount.toDouble()).toInt()
-            seekBar.progress = 0
+            // Fix "The stepSize must be 0, or a factor of the valueFrom-valueTo range" exception
+            seekBar.valueTo = widget.maxValue - (widget.maxValue - widget.minValue).rem(widget.step)
+            seekBar.valueFrom = widget.minValue
+            seekBar.stepSize = if (widget.step == 1F) 0F else widget.step
 
             val item = widget.item
             val state = item?.state ?: return
 
             if (item.isOfTypeOrGroupType(Item.Type.Color)) {
                 val brightness = state.asBrightness
                 if (brightness != null) {
-                    seekBar.max = 100
-                    seekBar.progress = brightness
+                    seekBar.valueFrom = 0F
+                    seekBar.valueTo = 100F
+                    seekBar.value = brightness.toFloat()
                 }
             } else {
                 val number = state.asNumber
                 if (number != null) {
-                    val progress = (number.value - widget.minValue) / widget.step
-                    seekBar.progress = Math.round(progress)
+                    var progress = number.value
+                    progress = min(progress, seekBar.valueTo)
+                    progress = max(progress, seekBar.valueFrom)
+                    seekBar.value = progress
```

Comment:
Can use this Kotlin method: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.ranges/coerce-in.html

Reply:
Done